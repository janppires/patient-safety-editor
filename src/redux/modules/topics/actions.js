import { normalize, schema } from 'normalizr';
export const FETCH_TOPICS = 'TOPICS/FETCH';
export const SET_SELECTED_TOPIC = 'TOPICS/SET_SELECTED';
export const GET_SELECTED_TOPIC = 'TOPICS/GET_SELECTED';

const topic = new schema.Entity('topics');
const topicsSchema = [ topic ];
const topics = [
  {id: 1, name: 'funny topic', points: [{}]},
  {id: 2, name: 'other', points: [{}]}
]

export const fetchTopics = () => {
  return {
    type: FETCH_TOPICS,
    payload: normalize(topics, topicsSchema)
  }
}

export const setSelectedTopic = (topicId) => {
  return {
    type: SET_SELECTED_TOPIC,
    payload: topicId
  }
}