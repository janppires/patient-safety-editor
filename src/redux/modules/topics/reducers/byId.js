import { FETCH_TOPICS, ADD_TOPIC } from "../actions";

export default function byId(state = {}, { type, payload }) {
  switch (type) {
    case FETCH_TOPICS:
      return payload.entities.topics;
    case ADD_TOPIC:
      return Object.assign({}, state, { [payload.id]: payload });
    default:
      return state;
  }
}
