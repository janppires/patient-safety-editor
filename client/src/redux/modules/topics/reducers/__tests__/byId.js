import byId from "../byId";
import { addTopic } from "../../actions";

describe("reducer: byId", () => {
  it("adds new topic by id to empty state", () => {
    const topic = { id: 2, name: "random name" };
    const action = addTopic(topic);
    expect(byId({}, action)).toMatchSnapshot();
  });

  it("adds new topic by id to existing state", () => {
    const state = prepareStateWithTopicsById();
    const topic = { id: 2, name: "random name" };
    const action = addTopic(topic);
    expect(byId(state, action)).toMatchSnapshot();
  });

  const prepareStateWithTopicsById = () => ({
    33: {
      id: 33,
      name: "other 33 topic"
    }
  });
});
