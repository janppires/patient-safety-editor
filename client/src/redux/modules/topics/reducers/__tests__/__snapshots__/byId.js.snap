// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer: byId adds new topic by id to empty state 1`] = `
Object {
  "2": Object {
    "id": 2,
    "name": "random name",
  },
}
`;

exports[`reducer: byId adds new topic by id to existing state 1`] = `
Object {
  "2": Object {
    "id": 2,
    "name": "random name",
  },
  "33": Object {
    "id": 33,
    "name": "other 33 topic",
  },
}
`;
